src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -651,32 +649,29 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
             }
         }
 
-        // Skips zero-initialization of the array if possible. If T contains object references,
-        // the array is always zero-initialized.
-        internal static T[] AllocateUninitializedArray<T>(int length)
+        /// <summary>
+        /// Skips zero-initialization of the array if possible. If T contains object references,
+        /// the array is always zero-initialized.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // forced to ensure no perf drop for small memory buffers (hot path)
+        internal static T[] AllocateUninitializedArray<T>(uint length)
```

Comment:
This should stay as `int`. `int` is in approved shape of the public API (https://github.com/dotnet/corefx/issues/31787). We just have not made this API public yet, and we do not want to be forking the API shape for internal use.

Reply:
Yes, the argument should be int.